qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/model.py

```diff
@@ -0,0 +1,572 @@
+                                         f"of space groups: \n\n {space_group_list} \n\n The phase file may need to be"
+                                         f"edited. For more information see: "
+                                         f"https://docs.mantidproject.org/nightly/concepts/PointAndSpaceGroups.html")
+
+        '''Validation'''
+        self.number_histograms = len(self.data_files)
+
+        if user_limits:
+            self.x_min = [float(k) for k in user_limits[0]]
+            self.x_max = [float(k) for k in user_limits[1]]
+
+        if self.x_min and self.x_max:  # limits have been set by the user
+            if len(self.x_min) != self.number_histograms:
+                if len(self.x_min) == 1:
+                    self.x_min = [self.x_min] * self.number_histograms
+                    self.x_max = [self.x_max] * self.number_histograms
+        else:
+            self.x_min = self.data_x_min
+            self.x_max = self.data_x_max
+            self.determine_x_limits()
+            if not self.x_min:
+                raise ValueError("Could not determine Minimum and Maximum X values from input files. "
+                                 + "Please set these values in the Plot Widget on the GSAS-II tab.")
+        limits = [self.x_min, self.x_max]
+
+        if self.refinement_method == 'Pawley' and not mantid_pawley_reflections:
+            raise ValueError(f"No Pawley Reflections were generated for the phases provided. Not calling GSAS-II.")
+
+        if len(self.instrument_files) != 1 and len(self.instrument_files) != self.number_histograms:
+            raise ValueError(f'The number of instrument files ({len(self.instrument_files)}) must be 1 '
+                             f'or equal to the number of input histograms {self.number_histograms}')
+
+        '''exec'''
+
+        gsas2_inputs = parse_inputs.Gsas2Inputs(
+            path_to_gsas2=path_to_gsas2,
+            temporary_save_directory=temporary_save_directory,
+            project_name=self.project_name,
+            refinement_method=self.refinement_method,
+            refine_background=refine_background,
+            refine_microstrain=refine_microstrain,
+            refine_sigma_one=refine_sigma_one,
+            refine_gamma=refine_gamma,
+            refine_histogram_scale_factor=refine_histogram_scale_factor,
+            data_files=self.data_files,
+            phase_files=phase_filepaths,
+            instrument_files=self.instrument_files,
+            limits=limits,
+            mantid_pawley_reflections=mantid_pawley_reflections,
+            override_cell_lengths=[float(x) for x in chosen_cell_lengths.split(" ")],
+            refine_unit_cell=refine_unit_cell,
+            d_spacing_min=self.dSpacing_min,
+            number_of_regions=self.number_of_regions
+        )
+
+        call_gsas2 = (path_to_gsas2 + "bin/python "
+                      + "/home/danielmurphy/mantid/qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/"
```

Comment:
The GSAS call currently fails for me because you've got this path which is specific to your machine. Can this be made a relative path?
Also I think the logic to append `path_to_gsas2` and bin\python could benefit from using os.path.join in case the user doesn't have a closing slash on the path in settings
Finally in my local GSAS II installation I don't have a bin directory. The python.exe is just sitting in the root directory of the install which in my case is `C:\Users\iao46725\gsas2full`. This is GSAS II v5203. Maybe that's not the right version...

Reply:
Fixed this and removed the used simply the gsas2_path + python for Windows